bpf/bpf_sock.c

```diff
@@ -677,6 +690,35 @@ static __always_inline int sock6_update_revnat(struct bpf_sock_addr *ctx,
 				      &val, 0);
 	return ret;
 }
+
+static __always_inline void ctx_get_v6_address(const struct bpf_sock_addr *ctx,
+					       union v6addr *addr);
+
+#ifdef ENABLE_LOCAL_REDIRECT_POLICY
+static __always_inline bool
+sock6_skip_xlate_from_ctx_to_svc(struct bpf_sock_addr *ctx __maybe_unused,
+				 const union v6addr address __maybe_unused,
+				 const __be16 port __maybe_unused)
+{
+#ifdef HAVE_NETNS_COOKIE
+	__net_cookie cookie = get_netns_cookie(ctx);
+	struct skip_lb6_key key;
+	__u8 *val = NULL;
+
+	memset(&key, 0, sizeof(key));
+	key.netns_cookie = cookie;
+	ctx_get_v6_address(ctx, &key.address);
```

Comment:
q: how come we pull the ip6 addr from the socket buffer here, whereas we use the passed ip4 address above?

Reply:
Ah! It's essentially the same address (the address is retrieved from the context and passed to this function), but we can directly use the passed address here as well. Thanks!